src/Microsoft.DotNet.Cli.Utils/Constants.cs

```diff
@@ -20,6 +20,7 @@ internal static class Constants
 
         public static readonly string HostExecutableName = "corehost" + ExeSuffix;
         public static readonly string DefaultConfiguration = "Debug";
+        public static readonly string DefaultPublishConfiguration = "Release";
```

Comment:
`Debug` must be the default option. `publish` is still part of development process and producing non-optimized debuggable bits makes it easier to debug issues in the dev cycle.

Similar to why VS produces debug builds by default.